client/src/components/RelatedProducts/RelatedProducts.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import API_KEY from '../../../../config.js';
+import RelatedProductCard from './RelatedProductCard.jsx';
+
+const RelatedProducts = ({ productId, handleClick, mainProductDetails }) => {
+
+  const [relatedProducts, getRelatedProducts] = useState([]);
+
+  useEffect(() => {
+    axiosGetRelatedProducts();
+  }, [productId]);
+
+
+  const axiosGetRelatedProducts = () => {
+    axios
+      .get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${productId}/related`, {
+        headers: {
+          'Authorization': API_KEY
+        }
+      })
+      .then((products) => getRelatedProducts(products.data))
+      .catch((error) => console.error(error));
+  };
+
+  const relatedCarousel = $(function() {
+    $('.jcarousel').jcarousel();
+  });
+
+  $('.jcarousel').jcarousel({
+    wrap: 'none'
+  });
+
+  let filter = { productId: 0 };
+
+  return (
+    <div className="wrapper">
+      <div className="jcarousel">
+        <ul>
+          {relatedProducts.map(relatedProductId => {
+            // eslint-disable-next-line eqeqeq
+            if (relatedProductId != productId && filter[relatedProductId] === undefined) {
+              filter[relatedProductId] = 1;
```

Comment:
We talked about this a little already. Try to implement the unique key attribute so it does not throw console errors. Also I would like a better understanding of why you are not using strict comparison here as well. If one is a number and the other is a string? Maybe convert the string into a number or vice versa.

Reply:
They were both coming back as typeof 'Number', so I was having an issue because I thought they were both numbers. Steve's solution seems to be working, though. As for the unique key, I can't seem to figure out a way to give them unique keys as using both index and product_id doesn't seem to do the trick.